Feat/json tree viewer

muzaffarmhd opened the pull request:
### PR Checklist

If you have any questions, you can refer to the [Contributing Guide](https://github.com/emqx/MQTTX/blob/main/.github/CONTRIBUTING.md)

#### What is the new behavior?
This Pull Request adds new features to the Viewer. Specifically follows the addition of Diff View with JSON Viewer.
Functionality wise the following features are added:
JSON Tree View: 
a. It allows user to view payloads in expandable and collapsible form using `vue-json-tree-view` as the dependency to parse and view JSON as a tree. Additionally, there is a `copy` option added which will copy the JSON payload to the system clipboard. If the payload is Non-JSON, it is displayed as a single collapsible node. 
Please check the corresponding UI:

<img width="1094" height="793" alt="image" src="https://github.com/user-attachments/assets/f792920b-52e5-41d3-9f51-c410532206a7" />

Additionally, there is also a search feature implemented to search in the payload.

<img width="693" height="680" alt="image" src="https://github.com/user-attachments/assets/7e7b8889-d921-4d41-a1f9-2d116344fd44" />

b. Visualization: We use ECharts to visualize the JSON as a tree, the depth level can be set, and on right click of node, the subtree will be copied as JSON. It also supports dark theme. 
<img width="1285" height="981" alt="image" src="https://github.com/user-attachments/assets/75720951-6025-4fc0-bd2d-0dca0157192d" />
c. Navigation: The payload history navigation is controlled through Arrow keys, Left and Right. 

#### Key Summary of utility function changes:
1. New utilities added:
- jsonTreeConverter: Converts a JSON structure to EChartsNode, the corresponding unit tests are added as well.
2. Modified utils:
- jsonUtils: new utility function added to stringify the ECharts subtree to JSON to avail copy functionality. Unit tests for this new utility are covered as well.

#### Does this PR introduce a breaking change?

- [ ] Yes
- [ x] No

ysfscream commented:
> Just realized, need to test and handle Big Int issues in JSON, which is common in IoT development. Should use the existing JSON Parse method instead of JS's JSON API

Can add some edge cases/boundary conditions to the test cases for testing

ysfscream commented on src/widgets/JSONViewer.vue:
```diff
@@ -0,0 +1,619 @@
+    return this.theme || 'light'
+  }
+
+  get treeOptions(): any {
```
It looks like there's a fixed format here, the type shouldn't be `any`?

muzaffarmhd replied:
Yes, my bad. I have fixed it. Thank you.

ysfscream commented on src/widgets/JSONViewer.vue:
```diff
@@ -0,0 +1,619 @@
+  private searchQuery = ''
+  private currentIndex = 0
+  private copiedJson = false
+  private searchTimeout: any = null
```
```suggestion
  private searchTimeout: number | null = null
```

ysfscream commented:
Overall I think there are no major issues, need to consider more edge cases, if there are performance issues should also consider them in advance, and there seem to be too many any types in the code

muzaffarmhd commented:
> > Just realized, need to test and handle Big Int issues in JSON, which is common in IoT development. Should use the existing JSON Parse method instead of JS's JSON API
> 
> Can add some edge cases/boundary conditions to the test cases for testing

I've added a utility to convert bigint instances to plain objects, if you think that is okay, i've added some test cases for it as well.

ysfscream commented:
> I've added a utility to convert bigint instances to plain objects, if you think that is okay, i've added some test cases for it as well.

It would be better if there were some test cases.
